### ISSUE TITLE
Check for missing handler for return false from function

### ISSUE THREAD

**Issue by 4n70w4** [kalessil/phpinspectionsea#1049]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Feature request                          |
| Plugin         | Php Inspections (EA Extended) |
| Language level | PHP 7.2                        |

Current behaviour (description/screenshot:)
--
nothing

Expected behaviour
--
warning message

Notes
---

Show warning notifications for functions that can return false in the case of an error and is no handled of false-case in code. Example: file_put_contents and imagejpeg returns false on error, but lack of verification can cause errors in other parts of the code.

```php
<?php

$image_file = 'test.jpg';

file_put_contents($image_file, 'some bites');

$image = imagecreatefromjpeg($image_file);

imagejpeg($image, $image_file);
```

Need warnings for file_put_contents($image_file, 'some bites'), imagejpeg($image, $image_file) and e.t.c. without check false case.

**Comment by kalessil** [comment]
So, if I getting this correctly, we are talking about un-checked file writes. Sounds like a good idea.

Here I'll request support by @4n70w4 , @rentalhost, @funivan: please prepare a list of such functions (similar to file_put_contents, imagejpeg) - I'm overloaded ATM.

**Comment by 4n70w4** [comment]
I'm talking about any functions that can return false in case of error (instead of exception), and the programmer did not check the result false.

**Comment by funivan** [comment]
@4n70w4 functions `json_encode`, `shuffle` and `preg_match` should be checked ? 

We can collect the list of functions and find how they are used in modern frameworks and tools (symfony, laravel, yii .....)

P.S. Google can help us:
1. `site:php.net "FALSE on failure" inurl:function`
2. `site:php.net "FALSE if an error occurred" inurl:function`

**Comment by 4n70w4** [comment]
Hi!

Yes, the volume of such functions is great. You can split the task into parts, for example:

1) the functions of the standard library, taking the information from here:
C:\Program Files\JetBrains\PhpStorm 2018.2.1\plugins\php\lib\php.jar -> \stubs\
2) a detector of similar functions written in user code

Patterns to detect the similar functions:
1) «false on failure» in return phpdoс (found 635 matches in PhpStorm\plugins\php\lib\php.jar -> \stubs\)
2) ...

**Comment by kalessil** [comment]
Search criteria are clear, still breakdown is needed. Covering 100% of those cases will emit lots of false-positives, hence I need a practical breakdown to start with: like files operations, string manipulation, regular expressions.

**Comment by kalessil** [comment]
So, after multiple talks at UGs, we are going to skip this request types. PhpStan is doing a good job here, so I don't see much benefits in replicating its' functionality.
